test.sh

```diff
@@ -1,40 +1,72 @@
-#!/bin/bash
+#!/usr/bin/env bash
+
+# -e Stops execution in the instance of a command or pipeline error
+# -u Treat unset variables as an error and exit immediately
+set -eu
+
 # Run all tests and coverage checks. Called from Travis automatically, also
 # suitable to run manually. See list of prerequisite packages in .travis.yml
 if type realpath >/dev/null 2>&1 ; then
-  cd $(realpath $(dirname $0))
+  cd "$(realpath -- $(dirname -- "$0"))"
 fi
 
-set -exo pipefail
-TRAVIS=${TRAVIS:-false}
+#
+# Defaults
```

Comment:
Document that the RUN, UNIT_FILTER, and INT_FILTER vars are here for backwards compatibility, and should be set to () unconditionally in the future.